plutarch-benchmark/bench/Main.hs

```diff
@@ -130,3 +139,226 @@ intListBench =
+                          PAddress addrFields <- pmatch x
+                          addrFields
+                      )
+                      # pconstant addr
+                , bench "sumtype(ignore-fields)" $
+                    plam
+                      ( \x -> P.do
+                          PMinting _ <- pmatch x
+                          pconstant ()
+                      )
+                      # pconstant minting
+                , bench "sumtype(partial-match)" $
+                    plam
+                      ( \x -> P.do
+                          PMinting hs <- pmatch x
+                          hs
+                      )
+                      # pconstant minting
+                , bench "sumtype(exhaustive)" $
+                    plam
+                      ( \x -> P.do
+                          purp <- pmatch x
+                          case purp of
+                            PMinting f -> plet f $ const $ phexByteStr "01"
+                            PSpending f -> plet f $ const $ phexByteStr "02"
+                            PRewarding f -> plet f $ const $ phexByteStr "03"
+                            PCertifying f -> plet f $ const $ phexByteStr "04"
+                      )
+                      # pconstant spending
+                , bench "sumtype(exhaustive)(ignore-fields)" $
+                    plam
+                      ( \x -> P.do
+                          purp <- pmatch x
+                          case purp of
+                            PMinting _ -> phexByteStr "01"
+                            PSpending _ -> phexByteStr "02"
+                            PRewarding _ -> phexByteStr "03"
+                            PCertifying _ -> phexByteStr "04"
+                      )
+                      # pconstant spending
+                ]
+            , benchGroup
+                "raw"
+                [ bench "newtype" $
+                    plam
+                      ( \x ->
+                          psndBuiltin #$ pasConstr # x
+                      )
+                      #$ pconstant
+                      $ toData addr
+                , bench "sumtype(ignore-fields)" $
+                    plam
+                      ( \x ->
+                          pif
+                            ((pfstBuiltin #$ pasConstr # x) #== 0)
+                            (pconstant ())
+                            perror
+                      )
+                      #$ pconstant
+                      $ toData minting
+                , bench "sumtype(partial-match)" $
+                    plam
+                      ( \x ->
+                          plet (pasConstr # x) $ \d ->
+                            pif
+                              (pfstBuiltin # d #== 0)
+                              (psndBuiltin # d)
+                              perror
+                      )
+                      #$ pconstant
+                      $ toData minting
+                , bench "sumtype(exhaustive)" $
```

Comment:
I *think* there's some inefficiency in this one and the "sumtype(exhaustive)(ignore-fields)" bench. Can't find it though.